### DISCUSSION TITLE
MPD not running    Error in /etc/mpd.conf line 25; '"' expected

### DISCUSSION THREAD

**Discussion by sp300b** [rern/rAudio#1004] [Fixed Issues]
Running rAudio-1 from image file rAudio-1-64bit-202207101.img working OK.
Here comes the MPD details:

mpd 0.23.8-1
max_output_buffer_size "12288"
audio_buffer_size      "6144"
bind_to_address        "0.0.0.0"
db_file                "/srv/http/data/mpd/mpd.db"
music_directory        "/mnt/MPD"
playlist_directory     "/var/lib/mpd/playlists"
replaygain             "off"
state_file             "/var/lib/mpd/mpdstate"
user                   "mpd"

input {
	plugin         "curl"
}

decoder {
	plugin         "ffmpeg"
	enabled        "yes"
}

resampler {
	plugin         "soxr"
	quality        "custom"
	precision      "32"
	phase_response "0"
	passband_end   "80"
	stopband_begin "105"
	attenuation    "0"
	flags          "32"
}

audio_output {
	name           "HiFiBerry DAC"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "none"
	format         "1536000:32:2" # custom
}

● mpd.service - Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: active (running) since Sun 2022-09-04 12:19:11 CEST; 15min ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
   Main PID: 404 (mpd)
      Tasks: 3 (limit: 1069)
        CPU: 1.257s
     CGroup: /system.slice/mpd.service
             └─404 /usr/bin/mpd --no-daemon

Sep 04 12:19:06 rAudio systemd[1]: Starting Music Player Daemon...
Sep 04 12:19:11 rAudio taskset[404]: decoder: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timidity.cfg
Sep 04 12:19:11 rAudio systemd[1]: Started Music Player Daemon.

But, when update 20220902 is executed, and followed by reboot, MPD can not be started any more.
The error data is provided as:

mpd 0.23.8-1
# cat /etc/mpd.conf
max_output_buffer_size "12288"
audio_buffer_size      "6144"
bind_to_address        "0.0.0.0"
db_file                "/srv/http/data/mpd/mpd.db"
music_directory        "/mnt/MPD"
playlist_directory     "/var/lib/mpd/playlists"
replaygain             "off"
state_file             "/var/lib/mpd/mpdstate"
user                   "mpd"

input {
	plugin         "curl"
}

decoder {
	plugin         "ffmpeg"
	enabled        "yes"
}

resampler {
	plugin         "soxr"
	quality        "custom"
	precision      "32"

audio_output {
	name           "HiFiBerry DAC"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "none"
	format         "1536000:32:2" # custom
}

mpd.service - Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: failed (Result: exit-code) since Sun 2022-09-04 12:38:19 CEST; 15s ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
    Process: 374 ExecStart=/usr/bin/taskset -c 3 /usr/bin/mpd --no-daemon (code=exited, status=1/FAILURE)
   Main PID: 374 (code=exited, status=1/FAILURE)
        CPU: 1.116s

Sep 04 12:38:13 rAudio systemd[1]: Starting Music Player Daemon...
Sep 04 12:38:18 rAudio taskset[374]: exception: Error in /etc/mpd.conf line 25; '"' expected
Sep 04 12:38:19 rAudio systemd[1]: mpd.service: Main process exited, code=exited, status=1/FAILURE
Sep 04 12:38:19 rAudio systemd[1]: mpd.service: Failed with result 'exit-code'.
Sep 04 12:38:19 rAudio systemd[1]: Failed to start Music Player Daemon.
Sep 04 12:38:19 rAudio systemd[1]: mpd.service: Consumed 1.116s CPU time.

**Comment by kosto-stavr** [comment]
same error after  update to 20220902  on raspberry pi 2 

`mpd.service - Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: failed (Result: exit-code) since Sun 2022-09-04 14:32:11 EEST; 7min ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
    Process: 975 ExecStart=/usr/bin/taskset -c 2,3 /usr/bin/mpd --no-daemon (code=exited, status=1/FAILURE)
   Main PID: 975 (code=exited, status=1/FAILURE)
        CPU: 1.576s

Sep 04 14:32:06 rAudio systemd[1]: Starting Music Player Daemon...
Sep 04 14:32:11 rAudio taskset[975]: exception: Error in /etc/mpd.conf line 31; '"' expected
Sep 04 14:32:11 rAudio systemd[1]: mpd.service: Main process exited, code=exited, status=1/FAILURE
Sep 04 14:32:11 rAudio systemd[1]: mpd.service: Failed with result 'exit-code'.
Sep 04 14:32:11 rAudio systemd[1]: Failed to start Music Player Daemon.
Sep 04 14:32:11 rAudio systemd[1]: mpd.service: Consumed 1.576s CPU time.`

**Comment by rern** [comment]
@sp300b, @kosto-stavr,

Sorry. There's a bug if SOXR Resampler is enabled.
Please try:
```sh
curl -L https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/settings/player-conf.sh > /srv/http/bash/settings/player-conf.sh
sed -i -e '/^\s*quality\|\s*precision/ d
' -e '/soxr/ a\
	quality        "very high"\
}
' /etc/mpd.conf

player-conf.sh
```
Then re-enable SOXR again.

**Reply by sp300b** [reply]
Bravo! Worked after applying this command



rAudio:~  #  curl -L https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/settings/player-conf.sh > /srv/http/bash/settings/player-conf.sh
sed -i -e '/^\s*quality\|\s*precision/ d
' -e '/soxr/ a\
        quality        "very high"\
}
' /etc/mpd.conf

player-conf.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  6081  100  6081    0     0   7641      0 --:--:-- --:--:-- --:--:--     0
rAudio:~  #
